server/src/main/java/org/candlepin/pinsetter/tasks/HypervisorUpdateJob.java

```diff
@@ -0,0 +1,270 @@
+
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.zip.DeflaterOutputStream;
+import java.util.zip.InflaterInputStream;
+
+/**
+ * Asynchronous job for refreshing the entitlement pools for specific
+ * {@link Owner}.
+ */
+public class HypervisorUpdateJob extends UniqueByOwnerJob {
+
+    private static Logger log = LoggerFactory.getLogger(HypervisorUpdateJob.class);
+    private OwnerCurator ownerCurator;
+    private ConsumerCurator consumerCurator;
+    private ConsumerResource consumerResource;
+
+    public static final String CREATE = "create";
+    public static final String DATA = "data";
+    public static final String PRINCIPAL = "principal";
+    protected static String prefix = "hypervisor_update_";
+
+    @Inject
+    public HypervisorUpdateJob(OwnerCurator ownerCurator, ConsumerCurator consumerCurator,
+            ConsumerResource consumerResource) {
+        this.ownerCurator = ownerCurator;
+        this.consumerCurator = consumerCurator;
+        this.consumerResource = consumerResource;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Executes {@link ConsumerResource#create(org.candlepin.model.Consumer, org.candlepin.auth.Principal,
+     *  java.utl.String, java.utl.String, java.utl.String)}
+     * Executes (@link ConusmerResource#performConsumerUpdates(java.utl.String, org.candlepin.model.Consumer)}
+     * as a pinsetter job.
+     *
+     * @param context the job's execution context
+     */
+    @Transactional
+    public void toExecute(JobExecutionContext context) throws JobExecutionException {
+        try {
+            JobDataMap map = context.getMergedJobDataMap();
+            String ownerKey = map.getString(JobStatus.TARGET_ID);
+            Boolean create = map.getBoolean(CREATE);
+            Principal principal = (Principal) map.get(PRINCIPAL);
+
+            HypervisorUpdateResult result = new HypervisorUpdateResult();
+
+            Owner owner = ownerCurator.lookupByKey(ownerKey);
+            if (owner == null) {
+                context.setResult("Nothing to do. Owner does not exist");
+                return;
+            }
+            byte[] data = (byte[]) map.get(DATA);
+            String json = decompress(data);
+            HypervisorList hypervisors = (HypervisorList) Util.fromJson(json, HypervisorList.class);
+            log.info("Hypervisor consumers for create/update: " + hypervisors.getHypervisors().size());
```

Comment:
Lets log the owner key here too for the sake of splunk debugging. "Updating hypervisor consumers for org {}", I might also log a quick note on the amount of data in the update.